test/Runtime/POSIX/Usage.c

```diff
@@ -0,0 +1,32 @@
+// This test checks for the (in)correct use of --sym-files and --sym-args
+
+// RUN: %llvmgcc %s -emit-llvm -O0 -c -o %t.bc
+// RUN: rm -rf %t.klee-out
+
+// RUN: not %klee --output-dir=%t.klee-out --exit-on-error --posix-runtime %t.bc --sym-files 1 10 --sym-files 2 10 &> %t1
+// RUN: grep "Multiple --sym-files are not allowed" %t1
```

Comment:
This should be FileCheck instead of grep.

Reply:
I like grep and only avoid it when FileCheck is nicer. How do you use FileCheck in a simple way for stderr?

Reply:
What I meant, why would FileCheck be better here, since there are no inline checks, line numbers, etc.? (using it with stderr is as easy as with stdout)

Reply:
I thought the idea of using `FileCheck` is because `grep` is not guaranteed to be installed.

I think I used `|&`  to pipe to FileCheck to check stderr.

Reply:
@ccadar Please use `FileCheck`. `grep`-based tools were deprecated by the LLVM developers with LLVM 3.3 (http://releases.llvm.org/3.3/docs/TestingGuide.html#id14). Beside that,  the error output in case something goes wrong is superior. As an example if something was changed, you get a detailled output (where it failed, what failed, what could be a potential match) this is nice for immediately seeing the issue in TravisCI for example.

In the following example, I changed "halting" to "haltung"
```
# command stderr:
klee/test/Feature/DumpStatesOnHalt.c:12:11: error: expected string not found in input
// CHECK: haltung execution, dumping remaining states
          ^
<stdin>:1:1: note: scanning from here
KLEE: output directory is "/test/Feature/Output/DumpStatesOnHalt.c.tmp.klee-out"
^
<stdin>:4:7: note: possible intended match here
KLEE: halting execution, dumping remaining states
      ^
```

Reply:
You are right, using `FileCheck` would add three lines of code and yes it would need prefixes and this test case is super-easy to read.
But, my reason to reduce the number of `grep` is to make it easier to write new test cases - especially for new contributors. How do you typically write a new test case? By copying and modifying an existing one!  
Many PRs tend to still use `grep` where a `FileCheck` is much more appropriate. They don't come up with using `grep`, they just copied and modified. They try to achieve complicated things that are much easier to express in `FileCheck`. Beside that, the `FileCheck` error messages are much more expressive.

Should `FileCheck` be used for everything? No, absolutely not (as seen in the other PR). `FileCheck` and `grep` have common features, I would rather like to advocate one over multiple as it suites our use-case better.

(Same for `clang-format`, we are not forcing it. Still, using it for guidance and automatic changes already improved the quality of the code base.)